ouroboros-network/src/Ouroboros/Network/Protocol/LocalStateQuery/Codec.hs

```diff
@@ -110,7 +128,10 @@ codecLocalStateQuery encodePoint  decodePoint
       case (stok, len, key) of
         (ClientAgency TokIdle, 2, 0) -> do
           pt <- decodePoint
-          return (SomeMessage (MsgAcquire pt))
+          return (SomeMessage (MsgAcquire (Just pt)))
+
+        (ClientAgency TokIdle, 1, 8) | canAcquireTip -> do
```

Comment:
Hmm. The negotiated version `V_8` means the sender was promised that `TokIdle, 1, 8` is valid message. So this `canAcquireTip` guard seems like we're potentially breaking that promise (I have not yet checked when exactly `canAcquireTip` is set to `False`.)

It's relatively new for me to be worrying about these serialization level concerns. Am I oversimplifying?

Reply:
Ah, nevermind. I see that the flag abstracts the negotiated version. The data flow makes sense 👍

Reply:
`canAcuireTip = version >= NodeToClientV_8`, where `version` is the negotiated version.  This means that these tokens are only valid for `V_8`.  There's even a [test](https://github.com/input-output-hk/ouroboros-network/pull/2875/commits/10ad3839e0895afed77b564b44f887174b4b7cf9#diff-ae122f3aa2edae6db73be6733912179ecb1d396e4bd973a38046766c6d874ab6R373) (its actually implemented in another [commit](https://github.com/input-output-hk/ouroboros-network/pull/2875/commits/77798f56b23816a6627752ff3e0710cb67ba4fb7) in this pr) which validates that the two codec versions are compatible.